morpho5/modules/vtk.morpho

```diff
@@ -0,0 +1,495 @@
+
+Contact: chaitanya.joshi@tufts.edu */
+
+import meshtools 
+import parser
+
+// Defining a VTK class that contains methods common to both the importer and the exporter
+class VTK {
+    init() {}
+
+    // Check whether `fieldname` is a a string is free of whitespaces (a VTK file requirement)
+    _checkfieldname(fieldname) {
+
+        // Check the fieldname
+        if (!isnil(fieldname)) {
+            // If not nil, first check whether the fieldname is a string
+            if (!isstring(fieldname)) {
+                self.err = Error("FnameNotStr", "Expected a string, but received `${fieldname}`.")
+                self.err.throw()
+            }
+            // Now that it's a string, checking for embedded whitespaces
+            for (c in fieldname) {
+                if (c==" ") {
+                    self.err = Error("InvalidFname", "fieldname (`${fieldname}`) cannot have embedded whitespaces.")
+                    self.err.throw()
+                }
+            }
+        }
+
+    }
+
+    
+    // Ensure that `filename` ends in ".vtk"
+    _ensurevtk(filename) {
+        
+        if (!isstring(filename)) {
+            self.err = Error("FnameNotStr", "Expected a string, but received `${filename}`.")
+            self.err.throw()
+        }
+        var fl = filename.split(".")
+
+        var n = fl.count()
+
+        if (n == 0) { // No extension
+            return "${filename}.vtk"
+        }
+        else if (fl[n-1] != "vtk") {
+            // Either a wrong extension has been provided, or the
+            // filename itself is expected to contain a period, like
+            // mesh.2.vtk. We again add ".vtk" at the end in this case.
+            return "${filename}.vtk"
+        }
+        else {
+            return filename
+        }
+
+    }
+
+}
+
+// VTKExporter: Exports Fields and / or Meshes to a .vtk file
+class VTKExporter is VTK {
+
+    init(obj, fieldname=nil) {
+        
+        // Initialize empty lists to optionally store fields later 
+        self.fields = []
+        self.fieldnames = []
+
+        // Check whether the object is a Mesh or a Field or none of them
+        if (ismesh(obj)) {
+
+            self.mesh = obj
+
+        } else if (isfield(obj)) {
+            
+            // If the object is a Field, then use its mesh for exporting
+            self.mesh = obj.mesh()
+            
+            // Add this field to the list
+            self.addfield(obj, fieldname=fieldname)
+
+        } else {
+            self.err = Error("InitErr", "Expected a Mesh or a Field object, but received `${obj}`.")
+            self.err.throw()
+        }
+
+        self.maxg = self.mesh.maxgrade()
+
+    }
+
+    // Write the preamble for the file. 
+    _preamble(file) {
+
+        file.write("# vtk DataFile Version 3.0 ") 
+        file.write("Exported using Morpho https://github.com/Morpho-lang/morpho ") // This line can be anything the user wants.
```

Comment:
// This line can be anything the user wants.

does the user have a way of specifying this?
or is the user here us?